### PR TITLE
Data and Server startup mappings

### PR DESCRIPTION
**Pull request by i509VCB**

### CODE REVIEW

File: mappings/net/minecraft/client/MinecraftClient.mapping [hunk 1]
```diff
@@ -252,3 +252,38 @@ CLASS net/minecraft/class_310 net/minecraft/client/MinecraftClient
 		ARG 1 sender
 	METHOD method_29043 isMultiplayerEnabled ()Z
 	METHOD method_29044 isOnlineChatEnabled ()Z
+	METHOD method_29599 (Lnet/minecraft/class_32$class_5143;Lnet/minecraft/class_5318$class_5319;Lnet/minecraft/class_3300;Lnet/minecraft/class_5359;)Lnet/minecraft/class_5219;
+		ARG 0 session
+		ARG 1 dimensionTracker
```
Comment (liach) [inline mappings/net/minecraft/client/MinecraftClient.mapping]: fyi i recently find out this is actually a registry tracker but written much like one for dimension (as proguard now reveals one more generic-setup method)

File: mappings/net/minecraft/client/MinecraftClient.mapping [hunk 2]
```diff
@@ -252,3 +252,38 @@ CLASS net/minecraft/class_310 net/minecraft/client/MinecraftClient
...
+		ARG 1 worldName
+		ARG 2 levelInfo
+		ARG 3 dimensionTracker
+		ARG 4 generatorOptions
+	METHOD method_29610 startIntegratedServer (Ljava/lang/String;Lnet/minecraft/class_5318$class_5319;Ljava/util/function/Function;Lcom/mojang/datafixers/util/Function4;ZLnet/minecraft/class_310$class_5366;)V
+		ARG 1 worldName
+		ARG 2 dimensionTracker
+		ARG 5 safemode
+	CLASS class_5366 WorldLoadAction
+	CLASS class_5367
```
Comment (liach) [inline mappings/net/minecraft/client/MinecraftClient.mapping]: Might name this thing

Comment (i509VCB) [reply mappings/net/minecraft/client/MinecraftClient.mapping]: I can't exactly figure out what it means. It seems to be a combo of ServerResourceManager, resource pack manager and another thing.

IntegratedResourceManager was my first thought

File: mappings/net/minecraft/resource/DataPackProfile.mapping [hunk 3]
```diff
@@ -0,0 +1,10 @@
+CLASS net/minecraft/class_5359 net/minecraft/resource/DataPackProfile
```
Comment (liach) [inline mappings/net/minecraft/resource/DataPackProfile.mapping]: more like data pack selection for a server? also not really resource related but more server related, as the lists' element types are like strings

File: mappings/net/minecraft/server/Main.mapping [hunk 4]
```diff
@@ -2,3 +2,21 @@ CLASS net/minecraft/server/Main
 	FIELD field_24625 LOGGER Lorg/apache/logging/log4j/Logger;
 	METHOD main ([Ljava/lang/String;)V
 		ARG 0 args
+	METHOD method_29734 (Lnet/minecraft/class_5318$class_5319;Lnet/minecraft/class_32$class_5143;Lnet/minecraft/class_3283;Lnet/minecraft/class_5350;Lnet/minecraft/class_5219;Lnet/minecraft/class_3807;Lcom/mojang/authlib/minecraft/MinecraftSessionService;Lcom/mojang/authlib/GameProfileRepository;Lnet/minecraft/class_3312;Ljoptsimple/OptionSet;Ljoptsimple/OptionSpec;Ljoptsimple/OptionSpec;Ljoptsimple/OptionSpec;Ljoptsimple/OptionSpec;Ljoptsimple/OptionSpec;Ljoptsimple/OptionSpec;Ljava/lang/Thread;)Lnet/minecraft/class_3176;
+		ARG 0 dimensionTracker
```
Comment (liach) [inline mappings/net/minecraft/server/Main.mapping]:
```suggestion
		ARG 0 registryTracker
```

File: mappings/net/minecraft/client/MinecraftClient.mapping [hunk 5]
```diff
@@ -252,3 +252,38 @@ CLASS net/minecraft/class_310 net/minecraft/client/MinecraftClient
...
+		ARG 5 sessionService
+		ARG 6 profileRepository
+		ARG 7 userCache
+		ARG 8 serverThread
+	METHOD method_29606 startIntegratedServer (Ljava/lang/String;)V
+		ARG 1 worldName
+	METHOD method_29607 (Ljava/lang/String;Lnet/minecraft/class_1940;Lnet/minecraft/class_5318$class_5319;Lnet/minecraft/class_5285;)V
+		ARG 1 worldName
+		ARG 2 levelInfo
+		ARG 3 dimensionTracker
```
Comment (liach) [inline mappings/net/minecraft/client/MinecraftClient.mapping]:
```suggestion
		ARG 3 registryTracker
```

File: mappings/net/minecraft/client/MinecraftClient.mapping [hunk 6]
```diff
@@ -252,3 +252,38 @@ CLASS net/minecraft/class_310 net/minecraft/client/MinecraftClient
 		ARG 1 sender
 	METHOD method_29043 isMultiplayerEnabled ()Z
 	METHOD method_29044 isOnlineChatEnabled ()Z
+	METHOD method_29599 createSaveProperties (Lnet/minecraft/class_32$class_5143;Lnet/minecraft/class_5318$class_5319;Lnet/minecraft/class_3300;Lnet/minecraft/class_5359;)Lnet/minecraft/class_5219;
+		ARG 0 session
+		ARG 1 dimensionTracker
+		ARG 2 resourceManager
+	METHOD method_29603 (Lnet/minecraft/class_5318$class_5319;Lnet/minecraft/class_32$class_5143;Lnet/minecraft/class_310$class_5367;Lnet/minecraft/class_5219;Lcom/mojang/authlib/minecraft/MinecraftSessionService;Lcom/mojang/authlib/GameProfileRepository;Lnet/minecraft/class_3312;Ljava/lang/Thread;)Lnet/minecraft/class_1132;
+		ARG 1 dimensionTracker
```
Comment (liach) [inline mappings/net/minecraft/client/MinecraftClient.mapping]:
```suggestion
		ARG 1 registryTracker
```

File: mappings/net/minecraft/client/MinecraftClient.mapping [hunk 7]
```diff
@@ -252,3 +252,38 @@ CLASS net/minecraft/class_310 net/minecraft/client/MinecraftClient
 		ARG 1 sender
 	METHOD method_29043 isMultiplayerEnabled ()Z
 	METHOD method_29044 isOnlineChatEnabled ()Z
+	METHOD method_29599 createSaveProperties (Lnet/minecraft/class_32$class_5143;Lnet/minecraft/class_5318$class_5319;Lnet/minecraft/class_3300;Lnet/minecraft/class_5359;)Lnet/minecraft/class_5219;
+		ARG 0 session
+		ARG 1 dimensionTracker
```
Comment (liach) [inline mappings/net/minecraft/client/MinecraftClient.mapping]:
```suggestion
		ARG 1 registryTracker
```

**Review by Juuxel — APPROVED** [review]

**Review by liach — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Comment by modmuss50** [comment]
Conflicts and then ready to merge.

**Review by YanisBft — APPROVED** [review]

**Pull request merged**